cohttp-async.opam

```diff
@@ -25,7 +25,7 @@ doc: "https://mirage.github.io/ocaml-cohttp/"
 bug-reports: "https://github.com/mirage/ocaml-cohttp/issues"
 depends: [
   "dune" {>= "3.8"}
-  "ocaml" {>= "4.14"}
+  "ocaml" {>= "4.14" & < "5.3.0"}
```

Comment:
what's the issue with OCaml 5.3 here?

Reply:
This is unfortunate but [`core_unix.v0.17` fails to build with OCaml 5.3](https://github.com/janestreet/core_unix/issues/14) but we also can't force an earlier version of `core_unix` as they all require OCaml <5.3, so [the CI then complains that there are no solutions for 5.3](https://github.com/mirage/ocaml-cohttp/actions/runs/13214152712/job/36891404090) (as it is specifically trying to test that version of OCaml, unless we tell it that <5.3 is expected). We would still hit this "no solution" issue if `core_unix.v0.17` was flagged as `ocaml<5.3` in the opam-repo, so I don't see how to lift that ocaml constraint until a new release of `core_unix` is published (but I'm open to suggestions!)